dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/OracleEvent.scala

```diff
@@ -0,0 +1,268 @@
+}
+
+sealed trait RangeV0OracleEvent extends OracleEvent {
+  override def eventDescriptorTLV: RangeEventDescriptorV0TLV
+  override def nonces: Vector[SchnorrNonce] = Vector(eventDescriptorTLV.nonce)
+}
+
+case class PendingRangeV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: RangeEventDescriptorV0TLV)
+    extends PendingOracleEvent
+    with RangeV0OracleEvent
+
+case class CompletedRangeV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: RangeEventDescriptorV0TLV,
+    attestation: FieldElement)
+    extends CompletedOracleEvent
+    with RangeV0OracleEvent {
+  override def attestations: Vector[FieldElement] = Vector(attestation)
+
+}
+
+sealed trait ExternalV0OracleEvent extends OracleEvent {
+  override def eventDescriptorTLV: ExternalEventDescriptorV0TLV
+  override def nonces: Vector[SchnorrNonce] = Vector(eventDescriptorTLV.nonce)
+}
+
+case class PendingExternalV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: ExternalEventDescriptorV0TLV)
+    extends PendingOracleEvent
+    with ExternalV0OracleEvent
+
+case class CompletedExternalV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: ExternalEventDescriptorV0TLV,
+    attestation: FieldElement)
+    extends CompletedOracleEvent
+    with ExternalV0OracleEvent {
+  override def attestations: Vector[FieldElement] = Vector(attestation)
+}
+
+sealed trait LargeRangeV0OracleEvent extends OracleEvent {
+  override def eventDescriptorTLV: LargeRangeEventDescriptorV0TLV
+
+  override def nonces: Vector[SchnorrNonce] = eventDescriptorTLV.nonces
+}
+
+case class PendingLargeRangeV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: LargeRangeEventDescriptorV0TLV)
+    extends PendingOracleEvent
+    with LargeRangeV0OracleEvent
+
+case class CompletedLargeRangeV0OracleEvent(
+    pubkey: SchnorrPublicKey,
+    eventName: String,
+    signingVersion: SigningVersion,
+    maturationTime: Instant,
+    announcementSignature: SchnorrDigitalSignature,
+    eventDescriptorTLV: LargeRangeEventDescriptorV0TLV,
+    attestations: Vector[FieldElement])
+    extends CompletedOracleEvent
+    with LargeRangeV0OracleEvent
+
+object OracleEvent {
+
+  def apply(eventDbs: Vector[EventDb]): OracleEvent = {
```

Comment:
nit: Maybe rename to `fromEventDbs()`? I always prefer using `.from()` rather tahn apply. I guess you can just create it and delegate directly the the `apply()` if you want.